hudi-common/src/main/java/org/apache/hudi/metadata/HoodieTableMetadataUtil.java

```diff
@@ -753,8 +752,8 @@ public static HoodieData<HoodieRecord> convertMetadataToColumnStatsRecords(Hoodi
           String partitionPath = deleteFileInfoPair.getLeft();
           String filePath = deleteFileInfoPair.getRight();
 
-          if (filePath.endsWith(HoodieFileFormat.PARQUET.getFileExtension()) || ExternalFilePathUtil.isExternallyCreatedFile(filePath)) {
-            return getColumnStatsRecords(partitionPath, filePath, dataMetaClient, columnsToIndex, true).iterator();
+          if (ExternalFilePathUtil.isExternallyCreatedFile(filePath)) {
+            return getColumnStatsRecords(partitionPath, filePath, dataMetaClient, columnsToIndex, true, false, Option.empty(), -1).iterator();
```

Comment:
maybe create a method to call this one that passes in the -1 since I see that call happens in multiple places. And it's a little iffy passing  -1